pep/001-initial-pep-process.md

```diff
@@ -0,0 +1,391 @@
+   and all pertinent concerns addressed. When the reviewer(s) agree that the
+   proposal is ready to implement, the pull request is merged. The status is now
+   **IMPLEMENTABLE**. 
+
+1. Author(s) submit a pull request to the relevant Porter repositories
+   implementing the proposal.
+
+1. Reviewer(s) evaluate the implementation. In addition to determining if the
+   implementation is acceptable, they should to identify gaps in the PEP, for
+   example the scope of change was much larger than originally thought, the
+   proposed implementation wasn't actually possible as specified or has
+   problems, etc.
+   
+   The pull request may be closed so the author(s) can go back and iterate on
+   the PEP. We should be careful about follow-up pull requests at this stage so
+   that a PEP isn't incompletely implemented. Critical changes should be made
+   together to limit follow-ups to unrelated changes that are not relevant to the PEP.
+
+1. After the implementation is merged, the owner submits a pull request to
+   change the status to `implemented`.
+
+1. Reviewer(s) confirm that the PEP has been completely implemented and no
+   additional work remains, and then merge the pull request. The final state is
+   now **IMPLEMENTED**.
+
+
+#### End States
+
+These are states that terminate the PEP process:
+
+* A PEP that successfully completes the proposal process is **IMPLEMENTED**.
+
+* The PEP may be rejected at certain points by the reviewer. Reviewer should
+  explain the rejection in the active pull request and ultimately close the pull
+  request unmerged. The reviewer submits a pull request updating the status to
+  `rejected` and reiterate the rejection, including enough context that someone
+  can understand it without going back to old pull requests. The status is
+  **REJECTED**.
+
+* The owner can withdraw the PEP by submitting a pull request changing the
+  status to **WITHDRAWN**.
+
+* When progress on the PEP ceases but we aren't ready to reject it, the PEP can
+  be deferred. For example when people are busy for months but they would
+  like to come back to it later or we want to do it but now isn't the right time
+  because other things need to be done first. In this case the owner or reviewer
+  can submit a pull request changing the status to **DEFERRED**.
+
+  When people are ready to work on it again, the owner should submit a pull
+  request changing the status back to an appropriate state to continue, such as
+  `provisional` or `implementable`.
+
+* Once a PEP has been merged, and a new PEP is started to replace it, or
+  supersede it, a pull request should be opened changing the status to
+  `replaced` and include a link to the new PEP. If the original PEP was
+  implemented, wait to update it to `replaced` until the new PEP is implemented.
+  For in-progress PEPs, they can be changed to `replaced` immediately. The
+  status is now **REPLACED**.
+
```

Comment:
Should there be a **RETIRED** state? for when something goes through the whole implementation process and then later it's removed? It could be possible that it's not that something should be replaced but it's no longer needed/necessary for whatever reason. 

Just thinking through the process, would a new PEP that removes something be "IMPLEMENTED" and then the old one would be"REPLACED" ?

Reply:
In the process that I described, deprecation/removal of something falls under "implementation". Implemented means the PEP as described as implemented. So if I had feature X, the first PEP would suggest adding feature X and eventually be marked implemented. Later when we changed our minds, a new PEP to remove the feature is proposed and once implemented (by removing the feature), the original PEP marked as replaced.

I can add a comment clarifying that it is for all changes, including removal of existing functionality.